packages/forms-web-app/src/dynamic-forms/options-question.js

```diff
@@ -129,31 +138,61 @@ class OptionsQuestion extends Question {
 	/**
 	 * returns the data to send to the DB
 	 * side effect: modifies journeyResponse with the new answers
-	 * @param {ExpressRequest} req
+	 * @param {import('express').Request} req
 	 * @param {JourneyResponse} journeyResponse - current journey response, modified with the new answers
-	 * @returns {Promise.<Object>}
+	 * @returns {Promise<{ answers: Record<string, unknown> }>}
 	 */
 	async getDataToSave(req, journeyResponse) {
```

Comment:
This is the bit that "fixes" not saving irrelevant data so we don't need all the nested ternaries in the BO formatters. In essence it just nullifies answers to conditional questions that the user should not have answered

Reply:
Could this be simplified to the following or would that miss something?

foreach opt in this.options
if opt in req.body -> set value
else -> set null

Reply:
Ah think I thought it was an issue because it's still in the DB, presumably it's an issue because it's still in the request.
Reckon it could still be simplified by doing null by default and then setting only the valid ones, rather than creating two arrays

Reply:
I can't think of a way to do this without making 2 arrays I'm afraid. To nullify everything I'd still need to construct an array of _all_ the conditionals so I can loop through that an nullify them. Then I'd loop through the valid ones. Am I missing something?

Reply:
```
		// sort conditional subsections they should be able to answer form those they shouldn't
		const [validConditionalFieldNames, allConditionalFieldNames] = this.options.reduce(
			(/** @type {[string[], string[]]} */ acc, option) => {
				if (!option.conditional) return acc;
				const optionIsSelectedOption = selectedOptions.some(
					(selectedOption) =>
						option.text === selectedOption.text && option.value === selectedOption.value
				);
				if (optionIsSelectedOption) acc[0].push(option.conditional.fieldName)
				return [acc[0], [...acc[1], option.conditional.fieldName]]
			},
			[[], []]
		);

		// nullify all fields
		allConditionalFieldNames.forEach((conditionalFieldName) => {
			const key = getConditionalFieldName(this.fieldName, conditionalFieldName);
			responseToSave.answers[key] = null;
			journeyResponse.answers[key] = null;
		});

		// add data from each valid conditional answer to data to be saved
		validConditionalFieldNames.forEach((validConditionalFieldName) => {
			const key = getConditionalFieldName(this.fieldName, validConditionalFieldName);
			const conditionalAnswer = req.body[key];
			if (!conditionalAnswer) return;
			responseToSave.answers[key] = req.body[key];
			journeyResponse.answers[key] = req.body[key];
		});
```

Reply:
was thinking something along the lines of a single loop instead of 3.

untested example:
```js
this.options.forEach((opt) => {
  if (!opt.conditional) return;
  
  const optionIsSelectedOption = selectedOptions.some(
    (selected) => opt.text === selected.text && opt.value === selected.value
  );
  
  const key = getConditionalFieldName(this.fieldName, opt.conditional.fieldName);
  const conditionalAnswer = req.body[key];
  
  if (!optionIsSelectedOption || !conditionalAnswer) {
    responseToSave.answers[key] = null;
    journeyResponse.answers[key] = null;
    return;
  }
  
  responseToSave.answers[key] = conditionalAnswer;
  journeyResponse.answers[key] = conditionalAnswer;
});
```

Shorter but potentially less explicit, happy to have it as is

Reply:
🤦‍♀️  right yeah sorry I get it, done